src/nrniv/nmodlrandom.cpp

```diff
@@ -0,0 +1,145 @@
+#include <../../nrnconf.h>
+
+/*
+HOC wrapper object for NMODL NEURON block RANDOM variables to give a HOC
+pointprocess.ranvar.method(...)
+sec.ranvar_mech(x).method(...)
+and Python
+poinprocess.ranvar.method(...)
+sec(x).mech.ranvar.method(...)
+syntax
+*/
+
+#include <section.h>
+#include <parse.hpp>
+#include <nrnran123.h>
+#include <classreg.h>
+#include <gui-redirect.h>
+
+struct NMODLRandom {
+    NMODLRandom(Object*) {}
+    ~NMODLRandom() {}
+    nrnran123_State* r() {
+        return (nrnran123_State*) hr_.get<void*>();
+    }
+    void chk() {
+        if (!prop_id_) {
+            hoc_execerr_ext("NMODLRandom wrapped handle is not valid");
+        }
+    }
+    neuron::container::generic_data_handle hr_{};
+    neuron::container::non_owning_identifier_without_container prop_id_{};
+};
+
+static Symbol* nmodlrandom_sym{};
+#undef dmaxuint
+#define dmaxuint 4294967295.
+
+static Object** set_ids(void* v) {  // return this NMODLRandom instance
+    NMODLRandom* r = (NMODLRandom*) v;
+    r->chk();
+    uint32_t id[3];
+    for (int i = 0; i < 3; ++i) {
+        id[i] = (uint32_t) (chkarg(i + 1, 0., dmaxuint));
+    }
+    nrnran123_setids(r->r(), id[0], id[1], id[2]);
+    return hoc_temp_objptr(nrn_get_gui_redirect_obj());
+}
+
+static Object** get_ids(void* v) {  // return a Vector of size 3.
+    NMODLRandom* r = (NMODLRandom*) v;
+    r->chk();
+    uint32_t id[3]{};
+    nrnran123_getids3(r->r(), id, id + 1, id + 2);
+    IvocVect* vec = vector_new1(3);
+    double* data = vector_vec(vec);
+    for (int i = 0; i < 3; ++i) {
+        data[i] = double(id[i]);
+    }
+    return vector_temp_objvar(vec);
+}
+
+static Object** set_seq(void* v) {  // return this NModlRandom instance
+    NMODLRandom* r = (NMODLRandom*) v;
+    r->chk();
+    double s = chkarg(1, 0., 17179869183.); /* 2^34 - 1 */
+    uint32_t seq = (uint32_t) (s / 4.);
+    char which = char(s - seq * 4.);
+    nrnran123_setseq(r->r(), seq, which);
```

Comment:
Perhaps I don't understand but the single arg version seems consistent with our discussion and the slack comment
> 5. Args choice  for set_seq() : (seq*4 + which)
the current syntax in HOC/Python is preferred as the user doesn't need to know the internal details of Random123 (seq vs which). We will change in NMODL so that we will also pass a single argument.ice  for set_seq() : (seq*4 + which)